pyproject.toml

```diff
@@ -75,6 +75,118 @@ requires = [
     "setuptools_scm>=7",
 ]
 
+[dependency-groups]
+build = [
+    "pybind11>=2.13.2,!=2.13.3",
+    "meson-python",
+    "setuptools-scm",
+]
+dev = [
+    {include-group= "build"},
+    {include-group = "doc"},
+    {include-group = "test"},
+    {include-group = "test-extra"},
+    {include-group = "lint"},
+]
+# Requirements for building docs
+#
+# You will first need a matching Matplotlib installation
+# e.g (from the Matplotlib root directory)
+#     pip install --no-build-isolation --editable .[dev]
```

Comment:
Is the `.[dev]` correct? Or should it be just `.`?

Reply:
The `[dev]` is an extra, not a dependency group. It was intended to cover both build and runtime dependencies. It didn't really work though, as `pip` installed `.` before its dependencies, so you couldn't get the build ones early enough. Based on the other implementations, dependency groups should be strictly better, since they should install before `.`, but we'll have to see how it's finally implemented. I should probably rewrite this section a bit once that's finalized.

Reply:
Oh, I see. I'm not used to seeing an extra be specified against a directory, so I thought it was a typo. And, yeah, extras wouldn't work for installing dependencies.